### DISCUSSION TITLE
proxy support for aws-s3-storage

### DISCUSSION THREAD

**Discussion by kongdewen** [verdaccio/verdaccio#4607] [Ideas]
**Is your feature request related to a problem?**
in our setup, verdaccio cannot connect to the internet directly. we need to support proxy option on s3 request.

**Describe the solution you'd like**
aws-sdk supports to pass the proxy option to the httpOptions setting of your config object as stated in https://github.com/aws/aws-sdk-js/blob/589d3b570e8cbf24976b7ce4606952be3363e2a0/doc-src/guide/Examples.md#using-proxies.
we would like to setup an option to pass the proxy to the s3 client as in https://github.com/verdaccio/monorepo/blob/main/plugins/aws-s3-storage/src/s3PackageManager.ts#L44.

**Describe alternatives you've considered**
adopt the proxy setting from Environment variable if presents

**Comment by mbtools** [comment]
adding a proxy option [here](https://github.com/verdaccio/monorepo/tree/main/plugins/aws-s3-storage#usage) sounds good. why don't you create a pull request with the suggested enhancement?

**Comment by juanpicado** [comment]
Probably also need it update AWS SDK https://github.com/verdaccio/verdaccio/pull/2574/files which has been tried already.

**Reply by kongdewen** [reply]
when is  the release of supporting aws sdk v3 scheduled to be out?

**Reply by juanpicado** [reply]
It's not on the roadmap to work on AWS sdk plugin upgrade so far, at least not me, it's too much work and I don't see happening this year, so I'd recommend someone else doing it.  The current SDK version is 1.x and is too old.  The same applies with the Google Cloud plugin.

**Comment by kongdewen** [comment]
FYI, A diff is created to support the proxy feature: https://github.com/verdaccio/monorepo/pull/713
